### ISSUE TITLE
Mistral Nemo Please!

### ISSUE THREAD

**Issue by stevengans** [ollama/ollama#5777]
https://mistral.ai/news/mistral-nemo/

**Comment by abenmrad** [comment]
I assume it's only Ollama team members that can publish models from the big players?

**Comment by stevengans** [comment]
There's licensing I believe?

**Comment by abenmrad** [comment]
According to the HF repo, it's Apache 2.0. But they require you to disclose the username and email (to Mistral?) before downloading from HF.

**Comment by SeanKnight** [comment]
https://huggingface.co/nvidia/Mistral-NeMo-12B-Instruct

**Comment by abenmrad** [comment]
There is indeed an Nvidia repo and Mistral repo (and only the latter requires the "disclosure").

Only the Mistral repo has safetensor files, which I'm now trying to quantize with ollama

**Comment by rick-github** [comment]
In case you are running up against `BPE pre-tokenizer was not recognized` errors, there are quantized versions at https://huggingface.co/second-state/Mistral-Nemo-Instruct-2407-GGUF/tree/main

**Comment by evannorstrand-mp** [comment]
When trying to convert the GGUF linked above, i get the following error:

Error: llama runner process has terminated: signal: aborted (core dumped) error loading model: check_tensor_dims: tensor 'blk.0.attn_q.weight' has wrong shape; expected  5120,  5120, got  5120,  4096,     1,     1

**Comment by rick-github** [comment]
Yeah, I get the same.  I guess that we need Nemo support in the llama.cpp GGUF converter.

**Comment by UmutAlihan** [comment]
let us patience a bit, it seems this nee model definitely be added to Ollama library since its popularity however it has been only a couple of hours I believe, so let us give it a humanely time to developers :)

**Comment by rick-github** [comment]
https://github.com/ggerganov/llama.cpp/issues/8577

**Comment by kiradzS** [comment]
I need it! immediately!

**Comment by enryteam** [comment]
I look forward to meeting it soon. come on, ollama

**Comment by phalexo** [comment]
> When trying to convert the GGUF linked above, i get the following error:
> 
> Error: llama runner process has terminated: signal: aborted (core dumped) error loading model: check_tensor_dims: tensor 'blk.0.attn_q.weight' has wrong shape; expected 5120, 5120, got 5120, 4096, 1, 1

I did NOT get this error converting. Converting went through without any error. The error happened when I tried to run the model, while loading.

**Comment by muhammadyusuf-kurbonov** [comment]
https://github.com/ggerganov/llama.cpp/pull/8579 is merged!

**Comment by rick-github** [comment]
@phalexo [evannorstrand-mp](https://github.com/evannorstrand-mp) means that when he tries to load the converted GGUF model, llama.cpp dumps core.  We are all experiencing that issue and are awaiting support in llama.cpp before we can load the model in ollama.

**Comment by droza123** [comment]
i think release b3426 of llama.cpp from yesterday adds support: https://github.com/ggerganov/llama.cpp/releases/tag/b3426

**Comment by rick-github** [comment]
b3426 adds the Tekken tokenizer, which Jeffrey merged into ollama in https://github.com/ollama/ollama/pull/5807.   However, that doesn't fix the dimensioning issue that causes the core dump when loading the Nemo model.  That's being addressed in https://github.com/ggerganov/llama.cpp/pull/8604 which looks close to being merged.

**Comment by stevengans** [comment]
an interesting quote on huggingface: 
> Unlike previous Mistral models, Mistral Nemo requires smaller temperatures. We recommend to use a temperature of 0.3

**Comment by LordFonDragon** [comment]
https://github.com/ggerganov/llama.cpp/pull/8604 IT MERGED!!!

**Comment by Qualzz** [comment]
MERGED YOUHOU

**Comment by UmutAlihan** [comment]
<img width="199" alt="image" src="https://github.com/user-attachments/assets/fdbc8b95-ed70-4eae-85e1-dcbc2b8b5f05">

**Comment by danilofalcao** [comment]
```
llama_model_load: error loading model: error loading model vocabulary: unknown pre-tokenizer type: 'tekken'
llama_load_model_from_file: failed to load model
Traceback (most recent call last):
  File "/root/Mistral-Nemo-Instruct-2407/chat.py", line 21, in <module>
    generated_text = run_inference(model_path, prompt)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/Mistral-Nemo-Instruct-2407/chat.py", line 5, in run_inference
    llm = Llama(model_path=model_path)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/env/lib/python3.11/site-packages/llama_cpp/llama.py", line 358, in __init__
    self._model = self._stack.enter_context(contextlib.closing(_LlamaModel(
                                                               ^^^^^^^^^^^^
  File "/root/env/lib/python3.11/site-packages/llama_cpp/_internals.py", line 54, in __init__
    raise ValueError(f"Failed to load model from file: {path_model}")
ValueError: Failed to load model from file: ggml-model-Q4_K_M.gguf
```

it's able to convert but still having issues with pre-tokenizer tekken using llama-cpp-python

**Comment by phalexo** [comment]
I rebuilt Ollama from the source this morning. The wrong dimension error on
loading is still there. Maybe it is because Ollama is using an
older version of llama.cpp.

**Comment by rick-github** [comment]
Yes, it still needs to be integrated into ollama.  Either via a patch like  https://github.com/ollama/ollama/pull/5807 or a rebase with llama.cpp.  In either case it will take a bit of time.

**Comment by ProjectMoon** [comment]
Looks like it's working as of 0.2.8-rc2! I Imported https://huggingface.co/bartowski/Mistral-Nemo-Instruct-2407-GGUF via this Modelfile below. Pretty sure I need a better template though.

```
FROM ./Mistral-Nemo-Instruct-2407-Q4_K_M.gguf
TEMPLATE """[INST] {{ if .System }}{{ .System }}{{ end }}{{ .Prompt }}[/INST] </s>"""
```

**Comment by stevengans** [comment]
Released!

**Comment by UmutAlihan** [comment]
Deeply appreciated for all the speed and effort from contributors <3

While searching the web I couldn't yet find a resource except function callilng through mistral_inference package.
I am curious whether anyone wrote a raw prompt in which function calling is working? 

I this a topic for another issue?

**Comment by asanchez75** [comment]
Thanks for the release! but I get this error `Error: llama runner process has terminated: signal: aborted (core dumped)` when I run `ollama run mistral-nemo:12b`. 

For your information, I have an NVIDIA GeForce RTX 4090 with 16G of GPU RAM.  Am I missing something?

**Comment by asanchez75** [comment]
I updated Ollama and it works! Thanks!

**Comment by phalexo** [comment]
I have tried Bartowski quant Q8_0 and it works fine, however, he does not
have the 16 bit version which was the one giving me the dimension error
before.
